### PR TITLE
Add user management for admins to the console

### LINKED ISSUE #1648: Add user management for admins to the console
#### Summary
Admin users should be able to manage other users via the console

#### Why do we need this?
To provide user management via a web interface (currently only possible via CLI)

#### What is already there? What do you see now?
Nothing of that sort.

#### What is missing? What do you want to see?
An admin area inside the console, where admins can configure users of the deployment. This includes:
- inviting a user
- listing users (API still needed)
- approving/suspending users

#### Environment
Console

#### How do you propose to implement this?
In the user dropdown (top right corner), we can add an "Admin Settings" section, which is only visible when the user has the admin flag. The user management could be a simple view, which lists all users (with search), all unapproved users and a button to invite users via email.
I'll do some quick wireframes for a POC.

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
Closes #1648 
References: #1086 

image

image

#### Changes
- Add user service to the JS SDK
- Add routing for admin settings
- Add views and containers for listing and editing users
- Extend feature checks for admin settings
- Extend redux primitives for listing, getting and updating users
- Minor related fixes such as PropTypes fixes

#### Notes for Reviewers
PR got quite big, but a lot of it is just plumbing and extending the redux primitives.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,7 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security
 
-## [3.3.2] (2019-12-04)
```
Comment (htdvisser) [inline CHANGELOG.md]: Undo. This was correct.

**Comment by htdvisser** [comment]
Is there a specific reason why "Users" isn't with the other "top level" entities? For admins, managing Users is just as important as managing Organizations.

**Comment by kschiffer** [comment]
> Is there a specific reason why "Users" isn't with the other "top level" entities? For admins, managing Users is just as important as managing Organizations.

We could display it in the top bar, but I wanted to group admin related views as part of the user dropdown. If we don't anticipate any other admin-only views in the future, it might indeed as well go into the top bar.

**Comment by johanstokking** [comment]
Setting priority of referenced issue

File: pkg/webui/console/components/user-data-form/index.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            danger
+            naked
+            message={sharedMessages.userDelete}
+            modalData={{
+              message: {
+                values: { orgName: initialValues.name || initialValues.ids.user_id },
+                ...m.modalWarning,
+              },
+            }}
+            onApprove={this.handleDelete}
```
Comment (bafonins) [inline pkg/webui/console/components/user-data-form/index.js]: `handleDelete` method is missing

File: pkg/webui/console/components/user-data-form/index.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+import Input from '../../../components/input'
+import Select from '../../../components/select'
+import SubmitBar from '../../../components/submit-bar'
+import SubmitButton from '../../../components/submit-button'
+import ModalButton from '../../../components/button/modal-button'
+
+import PropTypes from '../../../lib/prop-types'
+import sharedMessages from '../../../lib/shared-messages'
+
+const approvalStates = [
```
Comment (htdvisser) [inline pkg/webui/console/components/user-data-form/index.js]: Why are you not using the `enum:*` definitions from `message.json`?

**Pull request merged**